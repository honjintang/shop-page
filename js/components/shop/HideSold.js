import React from 'react'
import { connect } from 'react-redux'
import ReusableButton from '~/components/general/reusableButton'
import { toggleSoldItems } from '~/actions'
import { isShowingSoldItems } from '~/reducers/display'

export const HideSold = props => {
    const { toggleSoldItems, showSoldItems } = props

    return  (
        <ReusableButton 
            id='hide-show'
            label='Hide Sold Items'
            onClick={() => toggleSoldItems(!showSoldItems)}
            active={!showSoldItems}
        />
    )
}

const mapStateToProps = state => ({
    showSoldItems: isShowingSoldItems(state),
})

const mapDispatchToProps = {
    toggleSoldItems,
}

export default connect(mapStateToProps, mapDispatchToProps)(HideSold)