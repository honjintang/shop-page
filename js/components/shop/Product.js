import React from 'react'
import Card from 'react-bootstrap/Card'

const Product = ({id, img, title, brand, price, size, sold }) => (
        <Card id={id}>
            <Card.Img variant="top" src={img} />
            <Card.Body>
            <Card.Title>{title}</Card.Title>
            <Card.Text>
                <p>Brand: {brand || 'N/A'}</p>
                <p>Size: {size}</p>
                <p>Price: Â£{price}</p>
            </Card.Text>
            </Card.Body>
        </Card>
)

export default Product