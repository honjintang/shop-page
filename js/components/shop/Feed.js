import React from 'react'
import { connect } from 'react-redux'
import CardColumns from 'react-bootstrap/CardColumns'
import Product from '~/components/shop/Product'
import { getProducts } from '~/actions'
import { getAllProducts, getUnsoldProducts } from '~/reducers/products'
import { isShowingSoldItems } from '~/reducers/display'

export const Feed = props => {
    const { products, getProducts } = props
    React.useEffect(() => {
        getProducts()
    }, [getProducts])

    return  (
        <>
            {products &&
                <CardColumns>
                    {products.map(p => <Product key={p.id} {...p} />)}
                </CardColumns>
            }
        </>
    )
}

export const mapStateToProps = state => ({
    products: isShowingSoldItems(state) ? getAllProducts(state) : getUnsoldProducts(state),
})

const mapDispatchToProps = {
    getProducts,
}

export default connect(mapStateToProps, mapDispatchToProps)(Feed)
