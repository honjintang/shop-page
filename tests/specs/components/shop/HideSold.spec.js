import React from 'react'
import { shallow } from 'enzyme'
import { HideSold } from '~/components/shop/HideSold'

describe('components/shop/HideSold', () => {
    const mockToggleSoldItems= jest.fn()
    const defaultShowSoldItems = true
    let component

    beforeEach(() => {
        component = shallow(<HideSold toggleSoldItems={mockToggleSoldItems} showSoldItems={defaultShowSoldItems}/>)
    })

    it('matches snapshot', () => {
        expect(component).toMatchSnapshot()
    })

    it('toggles sold items when clicked', () => {
        expect(mockToggleSoldItems).toHaveBeenCalledTimes(0)
        component.simulate('click')
        expect(mockToggleSoldItems).toHaveBeenCalledTimes(1)
        expect(mockToggleSoldItems).toHaveBeenCalledWith(!defaultShowSoldItems)
    })
})