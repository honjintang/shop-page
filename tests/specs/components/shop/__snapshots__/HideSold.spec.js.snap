// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/shop/HideSold matches snapshot 1`] = `
<ReusableButton
  active={false}
  id="hide-show"
  label="Hide Sold Items"
  onClick={[Function]}
/>
`;
